src/conversions/chrono.rs

```diff
@@ -0,0 +1,663 @@
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for Utc {
+    fn extract(ob: &PyAny) -> PyResult<Utc> {
+        let py_tzinfo = <PyTzInfo as PyTryFrom>::try_from(ob)?;
+        let py_utc = timezone_utc(ob.py());
+        if py_tzinfo.eq(py_utc)? {
+            Ok(Utc)
+        } else {
+            Err(PyTypeError::new_err("Not datetime.timezone.utc"))
+        }
+    }
+}
+
+#[cfg(test)]
+mod test_chrono {
+    use crate::types::*;
+    use crate::{Python, ToPyObject};
+    use chrono::offset::{FixedOffset, Utc};
+    use chrono::{DateTime, Duration, NaiveDate, NaiveTime};
+    use std::cmp::Ordering;
+
+    #[test]
+    fn test_pyo3_timedelta_topyobject() {
+        use std::panic;
+
+        // Utility function used to check different durations.
+        // The `name` parameter is used to identify the check in case of a failure.
+        let check = |name: &'static str, delta: Duration, py_days, py_seconds, py_ms| {
+            Python::with_gil(|py| {
+                let delta = delta.to_object(py);
+                let delta: &PyDelta = delta.extract(py).unwrap();
+                let py_delta = PyDelta::new(py, py_days, py_seconds, py_ms, true).unwrap();
+                assert!(
+                    delta.eq(py_delta).unwrap(),
+                    "{}: {} != {}",
+                    name,
+                    delta,
+                    py_delta
+                );
+            });
+        };
+
+        let delta = Duration::days(-1) + Duration::seconds(1) + Duration::microseconds(-10);
+        check("delta normalization", delta, -1, 1, -10);
+
+        let delta = Duration::seconds(-86399999913600); // min
+        check("delta min value", delta, -999999999, 0, 0);
+
+        let delta = Duration::seconds(86399999999999) + Duration::nanoseconds(999999000); // max
+        check("delta max value", delta, 999999999, 86399, 999999);
+
+        // Also check that trying to convert an out of bound value panics.
+        Python::with_gil(|py| {
+            assert!(panic::catch_unwind(|| Duration::min_value().to_object(py)).is_err());
+            assert!(panic::catch_unwind(|| Duration::max_value().to_object(py)).is_err());
+        });
+    }
+
+    #[test]
+    fn test_pyo3_timedelta_frompyobject() {
+        // Utility function used to check different durations.
+        // The `name` parameter is used to identify the check in case of a failure.
+        let check = |name: &'static str, delta: Duration, py_days, py_seconds, py_ms| {
+            Python::with_gil(|py| {
+                let py_delta = PyDelta::new(py, py_days, py_seconds, py_ms, false).unwrap();
+                let py_delta: Duration = py_delta.extract().unwrap();
+                assert_eq!(py_delta, delta, "{}: {} != {}", name, py_delta, delta);
+            })
+        };
+
+        check(
+            "min value",
+            Duration::seconds(-86399999913600),
```

Comment:
Why not use `Duration::min_value()`?

Reply:
Because that's the min value allowed by python, `Duration::min_value()` is actually a lot smaller:
```
Duration::seconds(-86399999913600) = Duration { secs: -86399999913600, nanos: 0 }
Duration::min_value() =              Duration { secs: -9223372036854776, nanos: 192000000 }
```

Reply:
Why not do both rust min value and python min value? And leave a comment saying that one is rust min and one is python min.

Reply:
This is tested a few lines below that (currently 329-333), I just refactored the tests you wrote because it wasn't entirely clear what input data was making the test fail when they failed, but the assertions should not have changed.

Agree on the comment to make it clear why we are testing that specific value, I'll add it tomorrow once I work on it again